test/gateway/eip20_gateway/progress_revert_stake.js

```diff
@@ -0,0 +1,456 @@
+      1,
+      "Receipt status is unsuccessful",
+    );
+
+  });
+
+  it('should emit `StakeReverted` event ', async function () {
+
+    await gateway.setStorageRoot(revertStakeParams.blockHeight, revertStakeParams.storageRoot);
+
+    let tx = await gateway.progressRevertStake(
+      revertStakeParams.messageHash,
+      revertStakeParams.blockHeight,
+      revertStakeParams.rlpParentNodes,
+    );
+
+    let event = EventDecoder.getEvents(tx, gateway);
+    let eventData = event.StakeReverted;
+
+    assert.isDefined(
+      event.StakeReverted,
+      'Event `StakeReverted` must be emitted.',
+    );
+
+    assert.strictEqual(
+      eventData._messageHash,
+      revertStakeParams.messageHash,
+      `Message hash from the event must be equal to ${revertStakeParams.messageHash}.`,
+    );
+
+    assert.strictEqual(
+      eventData._staker,
+      revertStakeParams.staker,
+      `Staker address from the event must be equal to ${revertStakeParams.staker}.`,
+    );
+
+    assert.strictEqual(
+      eventData._stakerNonce.eq(revertStakeParams.nonce),
+      true,
+      `Staker nonce from the event must be equal to ${revertStakeParams.nonce}.`,
+    );
+
+    assert.strictEqual(
+      eventData._amount.eq(revertStakeParams.amount),
+      true,
+      `Stake amount from the event must be equal to ${revertStakeParams.amount}.`,
+    );
+
+  });
+
+  it('should burn bounty and penalty amount ', async function () {
+
+    let burnerAddress = await gateway.burner.call();
+
+    let gatewayInitialTokenBalance = await baseToken.balanceOf(gateway.address);
+    let stakerInitialTokenBalance = await baseToken.balanceOf(revertStakeParams.staker);
+    let burnerInitialTokenBalance = await baseToken.balanceOf(burnerAddress);
+
+    await gateway.setStorageRoot(revertStakeParams.blockHeight, revertStakeParams.storageRoot);
+
+    await gateway.progressRevertStake(
+      revertStakeParams.messageHash,
+      revertStakeParams.blockHeight,
+      revertStakeParams.rlpParentNodes,
+    );
+
+    let gatewayFinalTokenBalance = await baseToken.balanceOf(gateway.address);
+    let stakerFinalTokenBalance = await baseToken.balanceOf(revertStakeParams.staker);
+    let burnerFinalTokenBalance = await baseToken.balanceOf(burnerAddress);
+
+    assert.strictEqual(
+      gatewayFinalTokenBalance.eq(gatewayInitialTokenBalance.sub(bountyAmount.muln(2.5))),
+      true,
+      `Gateway balance must decrease by ${bountyAmount.muln(2.5).toString(10)}.`,
+    );
+
+    assert.strictEqual(
+      stakerInitialTokenBalance.eq(stakerFinalTokenBalance),
+      true,
+      `Staker balance must not change.`,
+    );
+
+    assert.strictEqual(
+      burnerFinalTokenBalance.eq(burnerInitialTokenBalance.add(bountyAmount.muln(2.5))),
+      true,
+      `Burner balance must increase by ${bountyAmount.muln(2.5).toString(10)}.`,
+    );
+
+  });
+
+  it('should transfer stake amount to staker ', async function () {
+
+    let burnerAddress = await gateway.burner.call();
```

Comment:
Why burner is important in this context?

Reply:
As there is a transfer of base token to burner address in revert progress. 
So it makes sense to assert the EIP20token balance to make sure that there is no transfer of EIP20 tokens to burner address.